### ISSUE TITLE
youtu.be short URLs don't work in Amara, Can we filter them?

### ISSUE THREAD

**Issue by jeremyclarke** [pculture/unisubs#2365]
From my testing any attempts to embed videos from YouTube using the http://youtu.be/XXX URLs fail completely, giving the message about there not being any HTML5 embed at that site. 

I'm talking about the /embeder-iframe/ system:
https://github.com/pculture/unisubs/wiki/Embed-Code-Usage-Guide

YouTube gives the youtu.be URLs as the default for sharing, so it seems like supporting them is a good idea, it's definitely confusing now since the error is about wrong site, rather than wrong URL format. 

In my WordPress/PHP code I used a simple filter to replace the youtu.be URLs with their youtube.com equivalents:

```
	/**
	 * Convert youtu.be/XXX links into youtube.com/watch?v=XXX
	 * Amara can't handle youtu.be
	 */
	if (strpos($url, 'youtu.be')) :
		$url = str_replace('youtu.be/', 'youtube.com/watch?v=', $url);
	endif;
```

I'm not really familiar with your setup but it seems like doing something similar would be just as easy on your end, and would solve the problem for a lot more people. 

In my tests it seems like all videos work in both formats as long as the XXX code for the video is in the "v" URL query. 

Thanks for all your hard work and hope this is helpful :)

**Comment by syl22-00** [comment]
@jeremyclarke thanks a lot for sharing yor suggestion. If you try to submit a shortened URL for a youtube video to add it on amara, it will be properly accepted, and its internal identifier will be its expanded URL. Then if you try to create an embedder for that video, the generated embedded code will include the correctly expanded URL.

That is to say, in a usual workflow, there should not be an issue for users related to those shortened URLs. Does that make sense?

**Comment by jeremyclarke** [comment]
Yes it does make sense, and it's good to know that they are handled well in the default workflow. 

It's funny because I can't really explain why my users would have ended up embedding a video with the youtu.be URL and marking it as an Amara video, but they did. I searched for the embed code we use for Amara ([embed unisubs='1']) and the first example had the youtu.be version of the URL. 

It's surely related to how we use the /embeder-iframe/ system to hot-convert plain YouTube URLs into Amara embeds, since users are way more likely to end up with a "fresh" version of the YouTube URL that they copy+paste long after the initial process of setting the video up for translation. 

That said it's kind of inherent to the embeder system that arbitrary YouTube URLs should work, so adding support for youtu.be URLs makes sense IMHO. This is especially true since it would mean that the two workflows are more similar, seeing as the default workflow does support youtu.be like you pointed out. 

Another use case would be if we were blanket-converting all YouTube embeds on our site to use Amara (e.g. encouraging visitors to transcribe/translate our videos). There again you'd get users embedding YouTube videos with the short URLs and it would be much better for it to work than for it to show that generic HTML5 error.
